web/src/beta/components/fields/TimelineField/index.tsx

```diff
@@ -0,0 +1,180 @@
+import { useCallback, useEffect, useState } from "react";
+
+import Button from "@reearth/beta/components/Button";
+import Icon from "@reearth/beta/components/Icon";
+import Text from "@reearth/beta/components/Text";
+import { useT } from "@reearth/services/i18n";
+import { styled } from "@reearth/services/theme";
+
+import Property from "..";
+
+import EditPanel from "./EditPanel";
+
+export type TimelineFieldProp = {
+  currentTime: string;
+  startTime: string;
+  endTime: string;
+};
+
+export type Props = {
+  name?: string;
+  description?: string;
+  value?: TimelineFieldProp;
+  onChange?: (value?: TimelineFieldProp) => void;
+};
+
+const TimelineField: React.FC<Props> = ({ name, description, value, onChange }) => {
+  const [openModal, setOpenModal] = useState(false);
+  const t = useT();
+
+  const handleTimelineModalCloser = useCallback(() => setOpenModal(false), []);
+
+  const handleTimelineModalOpener = useCallback(() => setOpenModal(true), []);
+
+  const handleRemoveSetting = useCallback(() => {
+    if (!value) return;
+    onChange?.();
+  }, [value, onChange]);
+
+  const [timelineValues, setTimelineValues] = useState(value);
+
+  useEffect(() => {
+    setTimelineValues(value);
+  }, [value]);
+
+  console.log("timelineValues.endTime", value);
+  return (
+    <Property name={name} description={description}>
+      <Wrapper>
+        <InputWrapper disabled={true}>
+          <Input dataTimeSet={!!timelineValues}>
+            <Timeline>
+              <TextWrapper size="footnote" customColor>
+                {timelineValues ? timelineValues.startTime : t("not set")}
```

Comment:
You can keep this kind of thing simpler by doing like this
`{timelineValues.startTime ?? t("not set")}`